### PR TITLE
Simplified the example

### PR DESCRIPTION
**Pull request by wiseoldman95**
"transfer-encoding: chunked\r\n" makes things a lot easier.

### CODE REVIEW

**Comment by wiseoldman95** [comment]
This seems to be causing weird issues, is transfer-encoding not supported?

File: examples/hello/hello.c [hunk 1]
```diff
@@ -6,22 +6,18 @@
...
               "HTTP/1.1 200 OK\r\n"
               "Content-Type: text/plain\r\n"
-              "Content-Length: %d\r\n"        // Always set Content-Length
-              "\r\n"
-              "%s",
-              content_length, content);
+              "transfer-encoding: chunked\r\n"
+              "\r\n");
 
+     mg_printf(conn, "Hello from civetweb! Remote port: %d", request_info->remote_port);
```
Comment (bel2125) [inline examples/hello/hello.c]: For chunked tranfer, you must not just send the data, but first you need to send the following chunk size (in hex).
e.g.

    int content_length = snprintf(content, sizeof(content),
                                  "Hello from civetweb! Remote port: %d",
                                  request_info->remote_port);
    mg_printf(conn, "%x\r\n", content_length);
    mg_printf(conn, "%s\r\n", content);  /* or better mg_write, especially for binary data */

then you must send an empty chunk to let the client know it's complete

    mg_printf(conn, "0\r\n\r\n");

By itself, mg_printf does not know if you are sending a header or a part of the body, and if this body needs chunked encoding or not.

**Comment by bel2125** [comment]
transfer-encoding: chunked from the server to the client is possible.
But you would need to send the chunk size manually - mg_printf does not do this automatically.
I added some comment to your example how it could be done (I did not test it).

transfer-encoding: chunked from the client to the server is something I plan to do some time later.

**Comment by wiseoldman95** [comment]
Thanks for explaining this.
Well, it *doesn't* make the example any simpler, so consider this pull request closed.

**Pull request closed without merging**